pkg/api/norman/customization/workload/workload.go

```diff
@@ -35,17 +39,30 @@ var (
 	allowRedeployTypes = map[string]bool{"cronJob": true, "deployment": true, "replicationController": true, "statefulSet": true, "daemonSet": true, "replicaSet": true}
 )
 
-type ActionWrapper struct {
+type WorkloadConfig struct {
 	ClusterManager *clustermanager.Manager
+	Schemas        map[string]*types.Schema
 }
 
-func (a ActionWrapper) ActionHandler(actionName string, action *types.Action, apiContext *types.APIContext) error {
+func (a *WorkloadConfig) ActionHandler(actionName string, action *types.Action, apiContext *types.APIContext) error {
 	var deployment projectclient.Workload
 	accessError := access.ByID(apiContext, &projectschema.Version, "workload", apiContext.ID, &deployment)
 	if accessError != nil {
 		return httperror.NewAPIError(httperror.InvalidReference, "Error accessing workload")
 	}
 	namespace, name := splitID(deployment.ID)
+
+	// Create a RawResource with a minimal config. This will be used for the eventual "CanDo" method, which parses
+	// the name and namespace from a RawResource object.
+	rawResource := &types.RawResource{
+		Values: map[string]interface{}{
+			"id":          name,
+			"namespaceId": namespace,
+		},
+	}
+	if a.canUpdateWorkload(apiContext, rawResource, getTypeFromAPIContext(apiContext)) != nil {
```

Comment:
We are missing some error handling here. `canUpdateWorkload` returns errors but they are never used. If the type/id is invalid a user will never know and get a 404. That isn't terribly useful. After calling `getTypeFromID` do a check on a.Schemas to validate it's a valid type, if not give back a real error. 
Or update splitID to give back all three pieces then do a check on the type here and ignore the type in other places

Reply:
Okay I can add that. If the type is invalid, is that worth returning an API error for?

Reply:
Actually, I do not know if we need to add extra error handling functionality given that we have the switch block handle whether or not the type is valid.

https://github.com/rancher/rancher/pull/29710/files/3fdf884e780f3d4c883b10c917209bcf6a061390#diff-78e6871c27609dec2cd98962aa49b9d15cb5dd9e35923aa279735c148aeefde9R325

What do you think about returning (bool, err) for this function, and having the ActionHandler return an API error based on the result? Example...

```go
if err := a.canUpdateWorkload(...); err != nil {
    if strings.Contains(err.Error(), "invalid workload type provided") {
        // return non-404 error
    } else {
        // return 404 error
    }
}

Reply:
We can also change this function to return (bool, error) as well. Thoughts?

Reply:
If you update canUpdateWorkload to return an error make it a typed error so you can just do a type check vs attempting to read the string

Reply:
Agree with that. Will do